server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -499,6 +498,16 @@ private static void nested(DocumentParserContext context, NestedObjectMapper nes
         }
     }
 
+    private static boolean parentIncludedInRoot(DocumentParserContext context, NestedObjectMapper nested) {
```

Comment:
I'm not sure this captures the same cases we did before. For example, you could have a triply-nested structure like this: 

```
{
  "properties": {
    "first": {
      "type": "nested",
      "include_in_parent": true,
      "properties": {
        "second": {
          "type": "nested",
          "include_in_parent": true,
          "properties": {
            "third": {
              "type": "nested",
              "include_in_parent": true,
              "include_in_root": true
            }
          }
        }
      }
    }
  }
}
```